share/hooks/ft/host_error.rb

```diff
@@ -283,6 +290,15 @@ def states_xpath(*arr)
             log "delete #{vm_id}"
             vm.delete
         when :migrate
+            vm_ds_id  = vm.retrieve_elements("/VM/HISTORY_RECORDS/HISTORY[position()=last()]/DS_ID")[0]
+
+            ds_xpath  = "/DATASTORE_POOL/DATASTORE[ID=\"#{vm_ds_id}\"]/TEMPLATE/SHARED"
+            is_shared = ds_pool.retrieve_elements(ds_xpath)[0]
+
+            if is_shared == "NO"
+                log "Skipping VM #{vm_id} deployed on non-shared datastore"
+                next
+            end
```

Comment:
So suggestion could be a rescue for this block that would skip the migration

Reply:
After looking at the code again, I agree the rescue seems like the cleanest way about this